provider/aws/sia-ec2/authn.go

```diff
@@ -100,13 +100,29 @@ func GetEC2Config(configFile, metaEndpoint string, useRegionalSTS bool, region,
 			if err != nil {
 				log.Printf("Unable to process security credentials: %v\n", err)
 				log.Println("Trying to determine service name from profile arn...")
-				configAccount, err = options.InitProfileConfig(metaEndpoint, "-service")
+				configAccount, err = options.InitProfileConfig(metaEndpoint, "-service", "-service@")
 				if err != nil {
-					log.Printf("Unable to determine service name: %v\n", err)
-					return config, nil, err
+					return config, nil, nil, err
 				}
 			}
 		}
 	}
-	return config, configAccount, nil
+
+	profileConfig, err := GetEC2AccessProfile(profileConfigFile, metaEndpoint)
```

Comment:
I'd like to see this incorporated into the InitFileConfig call and not have something specific.